rust/db_handling/src/identities.rs

```diff
@@ -96,6 +97,63 @@ lazy_static! {
     static ref REG_PATH: Regex = Regex::new(r"^(?P<path>(//?[^/]+)*)(///(?P<password>.+))?$").expect("known value");
 }
 
+#[derive(Encode, Decode)]
+pub enum ExportAddrs {
+    V1(ExportAddrsV1),
+}
+
+#[derive(Encode, Decode)]
+pub struct ExportAddrsV1 {
+    pub addrs: Vec<AddrInfo>,
+}
+
+#[derive(Encode, Decode)]
+pub struct AddrInfo {
+    /// Name of the key
+    pub name: String,
+
+    /// Address in the network.
+    ///
+    /// This is either `ss58` form for substrate-based chains or h160 form for ethereum based
+    /// chains
+    pub address: String,
```

Comment:
Shouldn't the public key be here also?

Reply:
As we discussed with the Ethereum keys case I see no need for UI to know the public key and to address things by public key. Public key "equals" `AddressId32` only for `sr25519` and `ed25519`, it does not for `ecdsa` that goes into `MultiSignature` and it does not for ethereum addrs.